tools/api_boost/api_boost.py

```diff
@@ -0,0 +1,159 @@
+  input_text = pathlib.Path(path).read_text()
+  for line in input_text.splitlines():
+    if include_lines and line.startswith('#include'):
+      output_lines.extend(include_lines)
+      include_lines = None
+    # Exclude API includes, except for a special case related to v2alpha
+    # ext_authz; this is needed to include the service descriptor in the build
+    # and is a hack that will go away when we remove v2.
+    if re.match(API_INCLUDE_REGEX, line) and 'envoy/service/auth/v2alpha' not in line:
+      continue
+    output_lines.append(line)
+
+  # Write to temporary file. We can't overwrite in place as we're executing
+  # concurrently with other ApiBoostFile() invocations that might need the file
+  # we're writing to.
+  pathlib.Path(path + TMP_SWP_SUFFIX).write_text('\n'.join(output_lines) + '\n')
+
+
+# Replace the original file with the temporary file created by ApiBoostFile()
+# for a given path.
+def SwapTmpFile(path):
+  pathlib.Path(path + TMP_SWP_SUFFIX).rename(path)
+
+
+# Update the Envoy source tree the latest API.
+def ApiBoostTree(args):
+  # Optional setup of state. We need the API type database, compilation database
+  # and api_booster tool in place before we can start boosting.
+  if args.generate_api_type_database:
+    sp.run(['./tools/gen_api_type_database.sh'], check=True)
+
+  if args.generate_compilation_database:
+    sp.run(['./tools/gen_compilation_database.py', '--run_bazel_build', '--include_headers'],
+           check=True)
+
+  if args.build_api_booster:
+    # Similar to gen_compilation_database.py, we only need the cc_library for
+    # setup. The long term fix for this is in
+    # https://github.com/bazelbuild/bazel/issues/9578.
+    dep_build_targets = [
+        '//source/...',
+        '//test/...',
+    ]
+    query = 'kind(cc_library, {})'.format(' union '.join(dep_build_targets))
+    dep_lib_build_targets = sp.check_output(['bazel', 'query', query]).decode().splitlines()
+    # Slightly easier to debug when we build api_booster on its own.
+    sp.run([
+        'bazel',
+        'build',
+        '--strip=always',
+        '@envoy_dev//clang_tools/api_booster',
+    ], check=True)
+    sp.run([
+        'bazel',
+        'build',
+        '--strip=always',
+    ] + dep_lib_build_targets, check=True)
+
+  # Figure out where the LLVM include path is. We need to provide this
+  # explicitly as the api_booster is built inside the Bazel cache and doesn't
+  # know about this path.
+  # TODO(htuch): this is fragile and depends on Clang version, should figure out
+  # a cleaner approach.
+  llvm_include_path = os.path.join(
+      sp.check_output([os.getenv('LLVM_CONFIG'), '--libdir']).decode().rstrip(),
+      'clang/9.0.0/include')
+
+  # Determine the files in the target dirs eligible for API boosting.
+  paths = []
+  for p in args.paths:
+    if os.path.isfile(p):
+      paths.append(p)
+      continue
+    for root, dirs, files in os.walk(p):
```

Comment:
just boost from compilation db because you're filtering it later anyway? not a fan of os.walk since it may accidentally include other stuff (e.g. indexer cache).